lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -162,9 +164,55 @@ define((require, exports, module) ->
     ch = ContextHandler = {
       contextPreviewExists: false
       UPDATE_DELAY: 1000
+      LABEL_SEQUENCE: [
+        {type: "keyword", value: /^\\label$/}
+        {type: "lparen", value: /^\{$/}
+        {type: "variable.parameter", value: /.*/}
+        {type: "rparen", value: /^\}$/}
+      ]
+      LABEL_PARAMETER_INDEX: 2
 
       getWholeEquation: (range) ->
-        editor.getSession().getTextRange(range)
+
+        tokenValues = []
+        labelSequenceIndex = 0
+        labelParameters = []
+        curLabelParameter = null
+        curLabelTokens = []
+
+        tokenIterator = new TokenIterator(editor.getSession(), range.start.row, range.start.column)
+        tokenIterator.stepForward() # at first it should be on the closing token of begin sequence
```

Comment:
Sry, did not understand your suggestion fully. To answer your actual question, I don't really see a point in creating token iterator before calling this function, and then passing it here. In every case we call it, we just pass it `ch.curRange`, and wrap the start of it every time in token iterator doesn't really seem reasonable to me.